pylops/basicoperators/pad.py

```diff
@@ -68,25 +70,25 @@ class Pad(LinearOperator):
 
     def __init__(
         self,
-        dims: Union[int, Tuple[int]],
-        pad: Tuple[int],
-        dtype: str = "float64",
+        dims: Union[int, InputDimsLike],
+        pad: Union[Tuple[int, int], Sequence[Tuple[int, int]]],
+        dtype: DTypeLike = "float64",
         name: str = "P",
     ) -> None:
         if np.any(np.array(pad) < 0):
             raise ValueError("Padding must be positive or zero")
-        dims = _value_or_list_like_to_tuple(dims)
+        dims = _value_or_sized_to_tuple(dims)
         # Accept (padbeg, padend) and [(padbeg, padend)]
-        self.pad = [pad] if len(dims) == 1 and len(pad) == 2 else pad
+        self.pad: Sequence = [pad] if len(dims) == 1 and len(pad) == 2 else pad
```

Comment:
Mmh what is the reasoning for annotating this. I think we create a lot of internal variables in init methods, but I am not sure we should go all the way to annotating all of them?

Reply:
Sometimes when annotating, the parser cannot tell exactly which type the variable will be. For example in this case, `self.pad` could be a `list` but if the user passes a `tuple`, it will be that. Mypy in this case will complain that it cannot figure out which type `self.pad` is. To get rid of the complaint, I annotated it.

My general approach to this PR was: if mypy gave me an error that I could fix without changing the code, I went ahead and did it.

Reply:
Ok, good point. Let's use this strategy. If we get an error and it is clear this is due to the parser having issues telling the type we annotate, otherwise we dont :)